src/helper/my-http-request.js

```diff
@@ -1,12 +1,25 @@
 var request  = require('request');
 
+var cache = {}
+
 /**
  * Creates an insecure (ignores certificate warnings) GET request for the given url
  *
  * @param {string} url URL of the request.
  * @param {function} callback Callback method for handling the response.
+ * @param {number} cacheInterval How long to cache the result for, or falsey to not cache
  */
-module.exports = function(url, callback) {
+module.exports = function(url, callback, cacheInterval) {
+    if (cacheInterval) {
+        var cached = cache[url];
+        if (cached && (Date.now() - cacheInterval) < cached.whenCached) {
+            setTimeout(function() {
```

Comment:
@datMaffin I used `setTimeout` to make sure the response callback was always asynchronous, as it would be if called from `request`.

That may not be necessary in practice here, and we could just try running without it. In general code can behave differently if a callback is sometimes called synchronously and sometimes asynchronously so I think it's more robust / safer now and in future. 

What do you think? If you agree I can add a comment.

Reply:
Interesting. Makes sense.

I see that using `setTimeout` would have been the cleaner approach (well, the code itself will look *less* clean 😉 ). 
The current codebase always called the callbacks in a blocking fashion.

I will just merge your commit as is and add a comment myself 🙂.